Kitodo/src/main/java/org/kitodo/production/services/file/FileService.java

```diff
@@ -1183,6 +1183,16 @@ private void addNewURIsToExistingPhysicalDivisions(Map<String, Map<Subfolder, UR
     private void addNewMediaToWorkpiece(List<String> canonicals, Map<String, Map<Subfolder, URI>> mediaToAdd,
             Workpiece workpiece) {
 
+        LogicalDivision actualLogicalRoot = workpiece.getLogicalStructure();
+        while (Objects.isNull(actualLogicalRoot.getType()) && actualLogicalRoot.getChildren().size() == 1) {
+            actualLogicalRoot = actualLogicalRoot.getChildren().get(0);
+        }
+        // If the newspaper has multiple issues in the process, then everything
+        // stays as it was
+        if (Objects.isNull(actualLogicalRoot.getType()) && actualLogicalRoot.getChildren().size() != 1) {
+            actualLogicalRoot = workpiece.getLogicalStructure();
```

Comment:
isn't this obsolete with line 1186?

Reply:
No. `actualLogicalRoot` is changed in line 1188, and this line sets it _back_ to what it was in line 1186 in case that the number of children is not 1 (that is, for processes with more than one newspaper issue; or (corrupted) processes with no issue at all).